torch/_inductor/kernel/flex_attention.py

```diff
@@ -871,14 +890,36 @@ def lower_cpu(
         )
 
     fake_buffers: List[Buffer] = []  # noqa: F821
+
+    # The value of cur_qSplitSize and cur_kvSplitSize are decided during runtime.
+    # We use symbols to represent them during the compilation here.
+    # They'll be replaced by the string "cur_qSplitSize" and "cur_kvSplitSize" in
+    # the modification function of the CppFlexAttentionTemplate class.
+    cur_qSplitSize = sympy.Symbol(
+        str(V.graph.sizevars.shape_env.create_unbacked_symint())
+    )
+    cur_kvSplitSize = sympy.Symbol(
+        str(V.graph.sizevars.shape_env.create_unbacked_symint())
+    )
+    shape_env = V.graph.sizevars.shape_env
+    assert cur_qSplitSize not in shape_env.var_to_range
+    assert cur_kvSplitSize not in shape_env.var_to_range
+
+    # We don't know the concret value of cur_qSplitSize and cur_kvSplitSize during the compilation.
+    # Mark symbols > 1 to ensure broadcasting is always applied.
+    # This avoids treating them as equal when `eq(var, 1)` is evaluated in `broadcast_symbolic_shapes`.
+    shape_env.var_to_range[cur_qSplitSize] = ValueRanges(2, int_oo)
+    shape_env.var_to_range[cur_kvSplitSize] = ValueRanges(2, int_oo)
```

Comment:
Is it possible `cur_qSplitSize` decided as 1 during runtime?

Reply:
In the generated code, we will always broadcast to the shape of the score tensor which is `[cur_qSplitSize, cur_kvSplitSize]`. The code will still be correct if `cur_qSplitSize` or `cur_kvSplitSize` is 1.